Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/service/decorator/HttpHeaderServiceEndpointDecorator.java

```diff
@@ -0,0 +1,144 @@
+import gov.hhs.fha.nhinc.properties.PropertyAccessor;
+import java.util.HashMap;
+import java.util.Set;
+import javax.xml.ws.BindingProvider;
+import org.apache.cxf.transports.http.configuration.ConnectionType;
+import org.apache.cxf.transports.http.configuration.HTTPClientPolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author jassmit
+ * @param <T>
+ */
+public class HttpHeaderServiceEndpointDecorator<T> extends ServiceEndpointDecorator<T> {
+
+    private final AssertionType assertion;
+
+    private static final Logger LOG = LoggerFactory.getLogger(HttpHeaderServiceEndpointDecorator.class);
+
+    public HttpHeaderServiceEndpointDecorator(ServiceEndpoint<T> decoratedEndpoint, final AssertionType assertion) {
+        super(decoratedEndpoint);
+        this.assertion = assertion;
+    }
+
+    @Override
+    public void configure() {
+        super.configure();
+        final HTTPClientPolicy httpClientPolicy = getHTTPClientPolicy();
+        final BindingProvider bindingProvider = (BindingProvider) getPort();
+
+        //Clear http connection value.
+        httpClientPolicy.setConnection(null);
+        if (hasKeepAliveForConnection()) {
+            httpClientPolicy.setConnection(ConnectionType.KEEP_ALIVE);
+        }
+
+        bindingProvider.getRequestContext().remove(NhincConstants.CUSTOM_HTTP_HEADERS);
+        HashMap<String, String> httpHeaders = getCustomHeaders();
+
+        if (httpHeaders != null && !httpHeaders.isEmpty()) {
+            bindingProvider.getRequestContext().put(NhincConstants.CUSTOM_HTTP_HEADERS, httpHeaders);
+        }
+    }
+
+    private boolean hasKeepAliveForConnection() {
+        String keepAlive = assertion.getKeepAlive();
+        if (NullChecker.isNullish(keepAlive)) {
+            keepAlive = getKeepAliveProperty();
+        }
+
+        return NullChecker.isNotNullish(keepAlive)
+                && ("TRUE".equalsIgnoreCase(keepAlive) || "T".equalsIgnoreCase(keepAlive));
+    }
+
+    private String getKeepAliveProperty() {
+        try {
+            return getPropertyAccessor().getProperty(NhincConstants.GATEWAY_PROPERTY_FILE, NhincConstants.KEEP_ALIVE_PROP);
+        } catch (PropertyAccessException ex) {
+            LOG.warn("Unable to access property: ", NhincConstants.KEEP_ALIVE_PROP, ex);
+            return null;
+        }
+    }
+
+    private HashMap<String, String> getCustomHeaders() {
+        HashMap<String, String> customHeaders = new HashMap<>();
+        try {
+            Set<String> allPropNames = getPropertyAccessor().getPropertyNames(NhincConstants.GATEWAY_PROPERTY_FILE);
+
+            if (NullChecker.isNotNullish(allPropNames)) {
+                addPropertiesToCustomHeaders(allPropNames, customHeaders);
+            }
+        } catch (PropertyAccessException ex) {
+            LOG.warn("Unable to access properties for custom http headers.", ex);
+        }
+
+        if (NullChecker.isNotNullish(assertion.getCONNECTCustomHttpHeaders())) {
```

Comment:
remove condition check since assertion.getCONNECTCustomHttpHeaders() will give empty ArrayList if connectcustomhttpHeader is null; ` public List<CONNECTCustomHttpHeadersType> getCONNECTCustomHttpHeaders() {
        if (connectCustomHttpHeaders == null) {
            connectCustomHttpHeaders = new ArrayList<CONNECTCustomHttpHeadersType>();
        }
        return this.connectCustomHttpHeaders;
    }`